hidimstat/estimator/_utils/u_Dnn_learner.py

```diff
@@ -0,0 +1,657 @@
+                else:
+                    curr_pred = relu(
+                        curr_pred.dot(current_model[3][grp_ind][ind_w_b])
+                        + current_model[4][grp_ind][ind_w_b]
+                    )
+            X_scaled_n[
+                :,
+                list(
+                    np.arange(input_dimensions[grp_ind], input_dimensions[grp_ind + 1])
+                ),
+            ] = (
+                curr_pred.dot(current_model[3][grp_ind][n_layer_stacking])
+                + current_model[4][grp_ind][n_layer_stacking]
+            )
+
+    n_layer = len(current_model[0]) - 1
+    for j in range(n_layer):
+        if j == 0:
+            pred = relu(X_scaled_n.dot(current_model[0][j]) + current_model[1][j])
+        else:
+            pred = relu(pred.dot(current_model[0][j]) + current_model[1][j])
+
+    pred = pred.dot(current_model[0][n_layer]) + current_model[1][n_layer]
+
+    if problem_type not in ("classification", "binary"):
+        if problem_type != "ordinal":
+            pred_v = pred * scaler_y.scale_ + scaler_y.mean_
+        else:
+            pred_v = activation_outcome[problem_type](pred)
+        loss = np.std(y_validation) ** 2 - mean_squared_error(
+            y_validation, pred_v[valid_ind]
+        )
+    else:
+        pred_v = activation_outcome[problem_type](pred)
+        loss = log_loss(
+            y_validation, np.ones(y_validation.shape) * np.mean(y_validation, axis=0)
+        ) - log_loss(y_validation, pred_v[valid_ind])
+
+    return (current_model, scaler_x, scaler_y, pred_v, loss)
+
+
+def _initialize_weights(layer):
+    if isinstance(layer, nn.Linear):
+        layer.weight.data = (layer.weight.data.uniform_() - 0.5) * 0.2
+        layer.bias.data = (layer.bias.data.uniform_() - 0.5) * 0.1
+
+
+def _dataset_Loader(X, y, shuffle=False, batch_size=50):
+    if y.shape[-1] == 2:
+        y = y[:, [1]]
+    dataset = torch.utils.data.TensorDataset(
+        torch.from_numpy(X).float(), torch.from_numpy(y).float()
+    )
+
+    loader = torch.utils.data.DataLoader(
+        dataset, batch_size=batch_size, shuffle=shuffle
+    )
+    return loader
+
+
+class DNN(nn.Module):
+    """
+    Feedfoward Neural Network with 4 hidden layers
+    """
+
+    def __init__(
+        self, input_dim, group_stacking, list_grps, output_dimension, problem_type
+    ):
+        super().__init__()
+        if problem_type == "classification":
+            self.accuracy = Accuracy(task="multiclass", num_classes=output_dimension)
+        else:
+            self.accuracy = Accuracy(task="binary")
+        self.list_grps = list_grps
+        self.group_stacking = group_stacking
+        if group_stacking:
+            self.layers_stacking = nn.ModuleList(
+                [
+                    nn.Linear(
+                        in_features=len(grp),
+                        out_features=input_dim[grp_ind + 1] - input_dim[grp_ind],
+                    )
+                    # nn.Sequential(
```

Comment:
Remove that stuff ?

Reply:
This will be addressed by the issue #80.